compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/DocumentationAnalyzer.java

```diff
@@ -228,7 +233,13 @@ public void visit(BLangTypeDefinition typeDefinition) {
                 validateReferences(field);
             }
         }
-        validateDeprecationDocumentation(typeDefinition.markdownDocumentationAttachment,
+
+        BLangMarkdownDocumentation documentation = typeDefinition.markdownDocumentationAttachment;
+        if (documentation != null && documentation.deprecatedParametersDocumentation != null) {
+            dlog.error(typeDefinition.pos, DiagnosticCode.DEPRECATED_PARAMETERS_DOCUMENTATION_NOT_ALLOWED);
```

Comment:
This still wouldn't detect scenarios like the following right?

```ballerina
# Docs for `i`
# 
# # Deprecated parameters
# + i - deprecated i
int i = 1;
```

We need to check all places where documentation is allowed.

Shall we work on a separate PR to fix this?

Reply:
+1

Reply:
@KavinduZoysa can you push the change for this too?

Reply:
This change is done